src/main/kotlin/org/rust/ide/inspections/fixes/AddTypeArguments.kt

```diff
@@ -0,0 +1,66 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.inspections.fixes
+
+import com.intellij.codeInspection.LocalQuickFixAndIntentionActionOnPsiElement
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiFile
+import org.rust.ide.inspections.getTypeArgumentsAndDeclaration
+import org.rust.lang.core.parser.RustParserUtil
+import org.rust.lang.core.psi.*
+import org.rust.lang.core.psi.ext.RsElement
+import org.rust.lang.core.psi.ext.RsGenericDeclaration
+import org.rust.lang.core.psi.ext.typeParameters
+import org.rust.openapiext.buildAndRunTemplate
+import org.rust.openapiext.createSmartPointer
+
+class AddTypeArguments(element: RsElement) : LocalQuickFixAndIntentionActionOnPsiElement(element) {
+    override fun getText(): String = "Add missing type arguments"
+    override fun getFamilyName() = text
+
+    override fun invoke(
+        project: Project,
+        file: PsiFile,
+        editor: Editor?,
+        startElement: PsiElement,
+        endElement: PsiElement
+    ) {
+        val element = startElement as? RsElement ?: return
+        val (typeArguments, declaration) = getTypeArgumentsAndDeclaration(element) ?: return
+
+        val requiredParameters = declaration.requiredGenericParameters
+        if (requiredParameters.isEmpty()) return
+
+        if (typeArguments?.typeReferenceList?.size ?: 0 >= requiredParameters.size) return
+
+        val factory = RsPsiFactory(project)
+
+        val argumentLifetimes = typeArguments?.lifetimeList?.map { it.text } ?: listOf()
+        val argumentTypes = typeArguments?.typeReferenceList?.map { it.text } ?: listOf()
```

Comment:
Do we really need `text` here? Maybe `name` is enough?

Reply:
Hmm, I thought that this would preserve comments, but actually even with `text` comments are lost. In any case, `name` is only available for lifetimes, if I'm not mistaken, type references can be of various types, so `text` has to be used? Or is there a way around it?